GNUmakefile

```diff
@@ -81,6 +81,8 @@ endif
 		GOOS=$(firstword $(subst _, ,$*)) \
 		GOARCH=$(lastword $(subst _, ,$*)) \
 		CC=$(CC) \
+		CGO_CFLAGS=$(CGO_CFLAGS) \
+		SDKROOT=$(SDKROOT) \
```

Comment:
This is safe, but we should almost certainly have something like the following at the top of the file to ensure they're unset by default but set from the user's environment if they do something like `CGO_FLAGS='-g -O2' make pkg/linux_amd64/nomad` (which is the kind of thing third-party packagers like Debian or Void might do):

```
CGO_CFLAGS ?= 
SDKROOT ?=
```

I'd also move `CGO_ENABLED` and `GO_LDFLAGS` close together so that all the compiler options are near each other when folks try to read this file.

Reply:
Thx for the feedback! 😸 I've added the defaults to the top.  

Also, before I push updates, did you mean move`CGO_ENABLED` and `CGO_CFLAGS` close together? Like this:

```shell
@CGO_ENABLED=$(CGO_ENABLED) \
		CGO_CFLAGS=$(CGO_CFLAGS) \
		GOOS=$(firstword $(subst _, ,$*)) \
		GOARCH=$(lastword $(subst _, ,$*)) \
		CC=$(CC) \
		SDKROOT=$(SDKROOT) \
		go build -trimpath -ldflags $(GO_LDFLAGS) -tags "$(GO_TAGS)" -o $(GO_OUT)
```

I didn't change anything with `GO_LDFLAGS` that I am aware of.

Reply:
Sorry, I meant the declarations outside of this target at [L9](https://github.com/hashicorp/nomad/blob/main/GNUmakefile#L9) and [L68](https://github.com/hashicorp/nomad/blob/main/GNUmakefile#L68). I realize `GO_LDFLAGS` didn't change, but I'm suggesting we have starting at line 9

```
GO_LDFLAGS := "-X github.com/hashicorp/nomad/version.GitCommit=$(GIT_COMMIT)$(GIT_DIRTY)"
CGO_ENABLED = 1
CGO_CFLAGS ?= 
SDKROOT ?=
```

so that all the user-set compiler flags are together. That's very much a nitpicky style idea though, so feel free to ignore 😁

Reply:
I love nitpicky! I'm on it.